rosbag2_storage_mcap/test/rosbag2_storage_mcap/test_mcap_storage.cpp

```diff
@@ -89,6 +102,24 @@ TEST_F(TemporaryDirectoryFixture, can_write_and_read_basic_mcap_file)
     auto reader = factory.open_read_only(expected_bag.string(), storage_id);
 #endif
     reader->open(options);
```

Comment:
@james-rms We have opened storage once in a few lines of code in ifdef section. 
Opening in one more time causing some undefined behavior for some reason and tests fails.
I think we should remove this extra `reader->open(options)`
```suggestion
```

Reply:
aah that's what I was missing. i was getting really confused :P